Mathlib/CategoryTheory/Bicategory/Kan/HasKan.lean

```diff
@@ -0,0 +1,308 @@
+/-
+Copyright (c) 2024 Yuma Mizuno. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuma Mizuno
+-/
+import Mathlib.CategoryTheory.Bicategory.Kan.IsKan
+
+/-!
+# Existence of Kan extensions and Kan lifts in bicategories
+
+We provide the propositional typeclass `HasLeftKanExtension f g`, which asserts that there
+exists a left Kan extension of `g` along `f`. See `CategoryTheory.Bicategory.Kan.IsKan` for
+the definition of left Kan extensions. Under the assumption that `HasLeftKanExtension f g`,
+we define the left Kan extension `lan f g` by using the axiom of choice.
+
+## Main definitions
+
+* `lan f g` is the left Kan extension of `g` along `f`, and is denoted by `f⁺ g`.
+* `lanLift f g` is the left Kan lift of `g` along `f`, and is denoted by `f₊ g`.
+
+## TODO
+
+* `ran f g` is the right Kan extension of `g` along `f`, and is denoted by `f⁺⁺ g`.
+* `ranLift f g` is the right Kan lift of `g` along `f`, and is denoted by `f₊₊ g`.
+
+-/
+
+noncomputable section
+
+namespace CategoryTheory
+
+namespace Bicategory
+
+universe w v u
+
+variable {B : Type u} [Bicategory.{w, v} B] {a b c : B}
+
+open Limits
+
+section LeftKan
+
+open LeftExtension
+
+variable {f : a ⟶ b} {g : a ⟶ c}
+
+/-- The existence of a left Kan extension of `g` along `f`. -/
+class HasLeftKanExtension (f : a ⟶ b) (g : a ⟶ c) : Prop where
+  hasInitial : HasInitial <| LeftExtension f g
+
+theorem LeftExtension.IsKan.hasLeftKanExtension {t : LeftExtension f g} (H : IsKan t) :
+    HasLeftKanExtension f g :=
+  ⟨IsInitial.hasInitial H⟩
+
+instance [HasLeftKanExtension f g] : HasInitial <| LeftExtension f g :=
+  HasLeftKanExtension.hasInitial
+
+/-- The left Kan extension of `g` along `f` at the level of structured arrows. -/
+def lanLeftExtension (f : a ⟶ b) (g : a ⟶ c) [HasLeftKanExtension f g] : LeftExtension f g :=
+  ⊥_ (LeftExtension f g)
+
+/-- The left Kan extension of `g` along `f`. -/
+def lan (f : a ⟶ b) (g : a ⟶ c) [HasLeftKanExtension f g] : b ⟶ c :=
+  (lanLeftExtension f g).extension
+
+/-- `f⁺ g` is the left Kan extension of `g` along `f`.
+```
+  b
+  △ \
+  |   \ f⁺ g
+f |     \
+  |       ◿
+  a - - - ▷ c
+      g
+```
+-/
+scoped infixr:90 "⁺ " => lan
```

Comment:
Again, is this a standard notation? (If so, could you add a reference to the literature in `docs/references.bib` and add it to the docstring of the file.)

Reply:
This notation is used in Kashiwara-Schapira, and I've added the references in the header docs. Although it is not very widely used I guess, I think it is an intuitive and convenient notation that's worth using.